src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_VALUE = 11;
+    private ColorSupplier randomColor = new ColorSupplier();
+    private Random random = new Random();
+    private String white = Color.WHITE.toString();
+    private int defaultRadius = 10;
+
+    public Figure getRandomFigure() {
+        Figure square = new Square(randomColor.getRandomColor(),
+                (double) random.nextInt(MAX_VALUE));
+        Figure circle = new Circle(randomColor.getRandomColor(),
+                (double) random.nextInt(MAX_VALUE));
+        Figure rectangle = new Rectangle(randomColor.getRandomColor(),
+                (double) random.nextInt(MAX_VALUE), (double) random.nextInt(MAX_VALUE));
+        Figure rightTriangle = new RightTriangle(randomColor.getRandomColor(),
+                (double) random.nextInt(MAX_VALUE), (double) random.nextInt(MAX_VALUE));
+        Figure isoscelesTrapezoid
+                = new IsoscelesTrapezoid(randomColor.getRandomColor(),
+                (double) random.nextInt(MAX_VALUE), (double) random.nextInt(MAX_VALUE),
+                (double) random.nextInt(MAX_VALUE));
+        Figure[] figures = new Figure[]{square, circle, rectangle,
+                rightTriangle, isoscelesTrapezoid};
+        int randomIndex = random.nextInt(figures.length);
+        return figures[randomIndex];

```

Comment:
Great approach, however I believe that we can consider something more optimal here. Think about `switch-case` approach, that will make us free from creating so many objects per each method call.